File: C/a010.c
#include <stdio.h>

int num, count;
char output[10000], * oPt;

int main()
{
	while (scanf(" %d", &num) == 1)
	{
		oPt = output;
		for (int i = 2; i <= num; i++)
		{
			while (!(num % i))
			{
				num /= i;
				count++;
			}
			if (count)
			{
				if (count == 1)
					oPt += sprintf(oPt, "%d * ", i);
				else
					oPt += sprintf(oPt, "%d^%d * ", i, count);
				count = 0;
			}
		}
		*(oPt - 3) = 0;
		puts(output);
	}
	return 0;
}

File: Python/a010.py
import sys

for fs in sys.stdin:
    num = int(fs.strip('\n'))
    out = ''
    i = 2
    while i <= num:
        count = 0
        while(num % i == 0):
            count += 1
            num /= i
        if count:
            if count == 1:
                out += str(i) + " * "
            else:
                out += str(i) + " ^ " + str(count) + " * "
        i += 1
    print(out[:-3])
